webapp/graphite/render/datalib.py

```diff
@@ -281,7 +281,11 @@ def prefetchData(requestContext, pathExpressions):
 
   if not requestContext.get('prefetched'):
     requestContext['prefetched'] = {}
-
-  requestContext['prefetched'][(startTime, endTime, now)] = prefetched
+  
+  if (startTime, endTime, now) in requestContext['prefetched']:
+    for k in prefetched:
+      requestContext['prefetched'][(startTime, endTime, now)][k] = prefetched[k]
```

Comment:
Rather than looping can we just use `update()`?
```suggestion
    requestContext['prefetched'][(startTime, endTime, now)].update(prefetched)
```

Reply:
Done.